waypoint/main.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import os
+from contextlib import asynccontextmanager
+
+from dependency_injector.wiring import Provide, inject
+from fastapi import Depends, FastAPI, HTTPException
+
+from shared.constants import PROJECT_VERSION
+from shared.log_config import get_logger
+from waypoint.routers import sse
+from waypoint.services.dependency_injection.container import Container
+from waypoint.services.nats_service import NatsEventsProcessor
+
+logger = get_logger(__name__)
+
+
+@asynccontextmanager
+async def app_lifespan(_: FastAPI):
+    logger.info("Waypoint Service startup")
+
+    container = Container()
+    await container.init_resources()
+
+    container.wire(modules=[__name__, sse])
+
+    await container.nats_events_processor()
+
+    yield
+
+    logger.debug("Shutting down Waypoint service...")
+    await container.shutdown_resources()
+    logger.info("Waypoint Service shutdown")
+
+
+def create_app() -> FastAPI:
+    openapi_name = os.getenv("OPENAPI_NAME", "Waypoint Service")
+
+    application = FastAPI(
+        title=openapi_name,
+        description="""
+        Welcome to the OpenAPI interface for the Waypoint Service.
+        The Waypoint Service enables the ability to wait for a specific event to occur before continuing.
+        """,
+        version=PROJECT_VERSION,
+        lifespan=app_lifespan,
+    )
+
+    application.include_router(sse.router)
+
+    return application
+
+
+logger.info("Waypoint Service startup")
+
+app = create_app()
+
+
+@app.get("/health/live")
+async def health_live():
+    return {"status": "live"}
+
+
+@app.get("/health/ready")
+@inject
```

Comment:
I'm noticing there's two health endpoints, when we probably only need one? In the waypoint logs I can see both get called every 10 seconds, which I don't understand because docker-compose healthcheck for waypoint just has: `test: ["CMD-SHELL", "curl -f http://localhost:3011/health/ready || exit 1"]`

Not sure - please ask Robbie or Willem what's best practice for health checks here

Reply:
In this case:

Liveness - if this passes, it means Uvicorn/FastAPI is correctly configured and everything is good to go.
If this fails, the Kubernetes Liveness Probe will trigger a restart.
If this happens too frequently too quickly, the container enters a CrashLoop.

Readiness - if this passes, it means the connection to NATS is functioning correctly and all is well.
If connection to NATS is lost or misconfigured, container will go into not-ready state.